ZichanYang324/BF-Employee-HR
# BGP-41 [Frontend and Backend] Registration link, email, and history

# Description
Implements BGP-41, generate registration token, send link to employee email, and show history

## Main changes explained:
- A new registration model
- A new JWT secret for registration token
- Hiring management controllers and router
- Hiring management page
- `ngx-toastr` as toaster
…

## How to test:
1. check into current branch, **add the following env credentials to `.env` of backend:**
```
JWT_EMAIL_SECRET=registrationemailtoken
EMAIL_USER=jackhallww@gmail.com
EMAIL_PASS='mcev cufc nsuv xcwi'
```
2. start the frontend and backend
3. check out `/hiring-management` on frontend
4. send an email then refresh the page to see the latest records

## Screenshots or videos of changes or expecting result:
![image](https://github.com/ZichanYang324/BF-Employee-HR/assets/85213835/ab0cc8b0-6943-4cf8-b18d-68d24e88e0b8)

![image](https://github.com/ZichanYang324/BF-Employee-HR/assets/85213835/7b5b0361-6379-46f2-b65d-9dc42e90f85e)


## Note:
The email will probably be in your trash or spam folder.
